bpdm-pool/src/main/kotlin/org/eclipse/tractusx/bpdm/pool/service/BusinessPartnerChangesDetector.kt

```diff
@@ -0,0 +1,259 @@
+ * Copyright (c) 2021,2023 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License, Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ******************************************************************************/
+
+package org.eclipse.tractusx.bpdm.pool.service
+
+import org.eclipse.tractusx.bpdm.pool.entity.*
+import org.springframework.stereotype.Service
+
+@Service
+class BusinessPartnerChangesDetector {
+    fun haveLegalEntityChanges(original: LegalEntity, updated: LegalEntity): Boolean {
+        return (
+                original.bpn != updated.bpn ||
+                        haveLegalFormChanges(original.legalForm, updated.legalForm) ||
+                        haveSetIdentifiersChanges(original.identifiers, updated.identifiers) ||
+                        haveSetStateChanges(original.states, updated.states) ||
+                        haveSetLogisticAddressChanges(original.addresses, updated.addresses) ||
+                        haveSetSiteChanges(original.sites, updated.sites) ||
+                        haveSetClassificationChanges(original.classifications, updated.classifications) ||
+                        haveLegalNameChanges(original.legalName, updated.legalName) ||
+                        haveLogisticAddressChanges(original.legalAddress, updated.legalAddress)
+                )
+    }
+
+
+    fun haveSetIdentifiersChanges(originalSet: MutableSet<LegalEntityIdentifier>, updatedSet: MutableSet<LegalEntityIdentifier>): Boolean {
+
+        if (originalSet.size != updatedSet.size) {
+            return true
+        }
+
+        for (originalEntity in originalSet) {
+            val updatedEntity = updatedSet.find { it == originalEntity }
+
+            if (updatedEntity == null || haveIdentifierChanged(originalEntity, updatedEntity)) {
+                return true
+            }
+        }
+
+        return false
+    }
+
+    fun haveIdentifierChanged(original: LegalEntityIdentifier, updated: LegalEntityIdentifier): Boolean {
+        return (
+                original.value != updated.value ||
+                        haveIdentifierTypeChanges(original.type, updated.type) ||
+                        original.issuingBody != updated.issuingBody
+                )
+    }
+
+    fun haveIdentifierTypeChanges(original: IdentifierType, updated: IdentifierType): Boolean {
+        return (
```

Comment:
I think in general the paranthesis are not needed for your return statement. Just a hint. If you feel you want them there that's fine with me too